### PR TITLE
 (PUP-10714) Remove master_used from report 

### PR DESCRIPTION
**Pull request by ciprianbadescu**
Remove master_used from report since it was replaced by server_used

### CODE REVIEW

**Comment by ciprianbadescu** [comment]
@joshcooper, do we want to remove master_used in Puppet 7?

**Comment by joshcooper** [comment]
I am leaning towards not removing it for puppet 7, because we added `server_used` in 6.19, so there won't be much time for consumers of the API to switch. That said I don't believe `master_used` is accessed by any platform or PE code. For example, the `puppetdb` report processor doesn't know about it: https://github.com/puppetlabs/puppetdb/blob/main/puppet/lib/puppet/reports/puppetdb.rb#L35. I don't think the orchestrator or console services uses it either. So that leaves custom report processors... So given that, maybe we should remove it. One thing in our favor is that `master_used` is nil when not using `server_list` (we reduces the usefulness of the entry), and PE only recently switched to always setting `server_list`. So 👍  from me to remove.

**Review by joshcooper — APPROVED** [review]

File: lib/puppet/transaction/report.rb [hunk 1]
```diff
@@ -326,7 +322,7 @@ def to_data_hash
     }
 
     # The following is include only when set
-    hash['master_used'] = hash['server_used'] = @server_used unless @server_used.nil?
+    hash['server_used'] = @server_used unless @server_used.nil?
```
Comment (joshcooper) [inline lib/puppet/transaction/report.rb:325]: Sorry I missed this, I think we should always support reading the old value (and setting `server_used`).

**Comment by joshcooper** [comment]
We should also create a separate ticket for the removal (so it doesn't appear that the removal happened in 6.19)

File: lib/puppet/transaction/report.rb [hunk 2]
```diff
@@ -260,8 +258,6 @@ def initialize_from_hash(data)
 
     if data['server_used']
       @server_used = data['server_used']
-    elsif data['master_used']
-      @server_used = data['master_used']
```
Comment (joshcooper) [inline lib/puppet/transaction/report.rb:264]: Sorry I missed this, I think we should always support reading the old `master_used` value, and setting `@server_used`. When writing a report, we should always set `server_used`.

**Review by joshcooper — APPROVED** [review]

**Pull request merged**